delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/auth/AuthTokenProvider.scala

```diff
@@ -0,0 +1,61 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.auth
+
+import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.kernel.cache.KeyValueStore
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.IOUtils
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.AuthToken
+import monix.bio.UIO
+
+import java.time.{Duration, Instant}
+
+/**
+  * Provides an auth token for the service account, for use when comunicating with remote storage
+  */
+trait AuthTokenProvider {
+  def apply(): UIO[Option[AuthToken]]
+}
+
+object AuthTokenProvider {
+  def apply(auth: Option[AuthenticateAs], keycloakAuthService: KeycloakAuthService): AuthTokenProvider = {
+    auth match {
+      case Some(authAs) => new CachingKeycloakAuthTokenProvider(authAs, keycloakAuthService)
+      case None         => new AnonymousAuthTokenProvider
+    }
+  }
+  def anonymousForTest: AuthTokenProvider = new AnonymousAuthTokenProvider
```

Comment:
Perhaps we don't need to say test here, I was just trying to communicate that this wasn't really for prod code